src/tlo/methods/enhanced_lifestyle.py

```diff
@@ -1979,3 +1979,72 @@ def apply(self, population):
                 key=_property,
                 data=flatten_multi_index_series_into_dict_for_logging(data)
             )
+
+        # ---------------------- log properties associated with WASH
+
+        # unimproved sanitation
+        # NOTE: True = no sanitation
+        li_no_clean_drinking_water = len(
+            df[df.li_unimproved_sanitation & df.is_alive & (df.age_years < 5)]
+        ) / len(df[df.is_alive & (df.age_years < 5)]
+                ) if len(df[df.is_alive & (df.age_years < 5)]) else 0
```

Comment:
You should use `sum(condition)` to calculate the number of individuals matching a condition. Using `len(df[condition])` takes a slice of the population dataframe, copying data unnecessarily. You can also get the under 5s & 5-15 year olds once. Prepare these at the top and use the condition throughout.

For example:

```suggestion
        under_5 = df.is_alive & (df.age_years < 5)
        between_5_and_15 = df.is_alive & (df.age_years.between(5, 15)
        
        li_no_clean_drinking_water = sum(df.li_unimproved_sanitation & under_5) / sum(under_5) if sum(under_5) else 0
```

And use the same `under_5` and `between_5_and_15` for the others.